anonymizer/src/main/java/gov/va/api/health/bulkfhir/anonymizer/ResourceBasedSyntheticData.java

```diff
@@ -0,0 +1,71 @@
+package gov.va.api.health.bulkfhir.anonymizer;
+
+import gov.va.api.health.dstu2.api.datatypes.HumanName;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+@Slf4j
+@Builder
+public class ResourceBasedSyntheticData implements SyntheticData {
+
+  private final Names names;
+
+  /**
+   * We synthesize Dates and DateTimes by using: month = previous month day = first of the month
+   * This technique is a one way transformation and can't be reversed.
+   */
+  @Override
+  public String synthesizeDate(String rawDate) {
+    if (StringUtils.isBlank(rawDate)) {
+      return null;
+    }
+    LocalDate date;
+    try {
+      date = LocalDate.parse(rawDate);
+      date = date.withMonth(date.getMonthValue() - 1);
```

Comment:
date's should truncate to the _year_ I believe? @eclendenning-va

Reply:
"All elements of dates (except year) for dates directly related to an individual, including birth date, admission date, discharge date, date of death; and all ages over 89 and all elements of dates (including year) indicative of such age, except that such ages and elements may be aggregated into a single category of age 90 or older."